lightning/src/util/persist.rs

```diff
@@ -243,10 +250,889 @@ where
+				}
+			},
+			Err(ref e) if e.kind() == io::ErrorKind::NotFound => {},
+			_ => return chain::ChannelMonitorUpdateStatus::InProgress
+		}
+		// Serialize and write the new monitor
+		let mut monitor_bytes = Vec::with_capacity(
+			MONITOR_UPDATING_PERSISTER_PREPEND_SENTINEL.len() + monitor.serialized_length(),
+		);
+		monitor_bytes.extend_from_slice(MONITOR_UPDATING_PERSISTER_PREPEND_SENTINEL);
+		monitor.write(&mut monitor_bytes).unwrap();
+		match self.kv_store.write(
+			CHANNEL_MONITOR_PERSISTENCE_NAMESPACE,
+			CHANNEL_MONITOR_PERSISTENCE_SUB_NAMESPACE,
+			key,
+			&monitor_bytes,
+		) {
+			Ok(_) => {
+				// Assess cleanup
+				if let Ok((_, old_monitor)) = maybe_old_monitor {
+					// Typically, we'll clean up between the last two known full monitors
+					let start = old_monitor.get_latest_update_id();
+					let mut end = monitor.get_latest_update_id();
+					if monitor.get_latest_update_id() == CLOSED_CHANNEL_UPDATE_ID {
+						// We don't want to clean the rest of u64, so just do possible pending updates
+						end = start.saturating_add(self.maximum_pending_updates);
+					}
+					if let Err(e) =
+						self.lazy_delete_updates(&MonitorName::from(funding_txo), start, end)
+					{
+						log_error!(
+							self.logger,
+							"error cleaning up channel monitor updates {}",
+							e
+						);
+					};
+				};
+				chain::ChannelMonitorUpdateStatus::Completed
+			}
+			Err(e) => {
+				log_error!(self.logger, "error writing channel monitor {}", e);
+				chain::ChannelMonitorUpdateStatus::InProgress
+			}
+		}
+	}
+
+	/// Persists a channel update, writing only the update to the provided [`KVStore`] if possible.
+	///
+	/// In some cases, this will forward to [`MonitorUpdatingPersister::persist_new_channel`]:
+	///
+	///   - No full monitor is found in [`KVStore`]
+	///   - The number of pending updates exceeds [`self.maximum_pending_updates`]
+	///   - LDK commands re-persisting the entire monitor through this function, specifically when
+	///     `update` is `None`.
+	///   - The update is at [`CLOSED_CHANNEL_UPDATE_ID`]
+	fn update_persisted_channel(
+		&self,
+		funding_txo: OutPoint,
+		update: Option<&ChannelMonitorUpdate>,
+		monitor: &ChannelMonitor<ChannelSigner>,
+		monitor_update_call_id: MonitorUpdateId,
+	) -> chain::ChannelMonitorUpdateStatus {
+		// IMPORTANT: update_id: [`MonitorUpdateId`] is not to
+		// be confused with [`ChannelMonitorUpdate`]'s
+		// update_id.
+		if let Some(update) = update {
+			if update.update_id != CLOSED_CHANNEL_UPDATE_ID && update.update_id % self.maximum_pending_updates != 0 {
+				let ns = match MonitorName::try_from(funding_txo) {
+					Ok(monitor_name) => monitor_name.0,
+					Err(_) => return chain::ChannelMonitorUpdateStatus::InProgress,
+				};
+				let update_name = match UpdateName::try_from(update.update_id) {
+					Ok(update_name) => update_name,
+					Err(_) => return chain::ChannelMonitorUpdateStatus::InProgress,
+				};
```

Comment:
Can use `from` for both of these, too.

Reply:
done, still asking myself how I ended up with try_from there.